doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+.. currentmodule:: sklearn
+
+.. _glossary:
+
+=========================================
+Glossary of Common Terms and API Elements
+=========================================
+
+This glossary hopes to definitively represent the tacit and explicit
+conventions applied in Scikit-learn and its API, while providing a reference
+for users and contributors. It aims to describe the concepts and either detail
+their corresponding API or link to other relevant parts of the documentation
+which do so.
+
+General Concepts
+================
+
+.. glossary::
+
+    API
+        The specific interfaces for estimators implemented in Scikit-learn and
+        the generalized conventions across types of estimators as described in
+        this glossary.
+
+        The specific interfaces that constitute Scikit-learn's public API are
+        largely documented in :ref:`api_ref`. However we less formally consider
+        anything as public API if none of the identifiers required to access it
+        begins with ``_``.  We generally try to maintain :term:`backwards
```

Comment:
Two spaces after a full stop... your typewriter having issues? ;) Or did you transition to law?

Reply:
Oh my Dad by habit *always* puts two spaces after full stop...

Reply:
You did it relatively consistently in this PR ;) It was standard in the US 50 years ago (maybe in Australia too?). Lawyers in the US are also required to do it for any legal document, a friend told me.

Reply:
Do you care for me to fix them? I think they're artefacts of my line-wrapping macro in Vim. Maybe there's a setting somewhere....